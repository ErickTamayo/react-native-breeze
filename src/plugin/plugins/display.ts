import { PluginFunction } from "./types";

export type PluginGroups = { display: "hidden" | "flex" };

export const pattern = /^(?<display>hidden|flex)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { display } = groups;
  return { display: display === "hidden" ? "none" : "flex" };
};
