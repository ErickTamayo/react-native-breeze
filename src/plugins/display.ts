import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = { display: "hidden" | "flex" };

export const pattern: PluginPattern = /^(?<display>hidden|flex)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { display } = groups;
  return { display: display === "hidden" ? "none" : "flex" };
};
