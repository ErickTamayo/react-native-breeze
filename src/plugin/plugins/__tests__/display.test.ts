import {
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../utils/tests";
import { pattern, plugin } from "../display";

describe("display", () => {
  const shouldMatch = ["hidden", "flex"];

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
