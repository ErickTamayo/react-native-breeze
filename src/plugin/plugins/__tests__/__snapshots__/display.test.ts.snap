// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display Should parse (flex) correctly 1`] = `
Object {
  "display": "flex",
}
`;

exports[`display Should parse (hidden) correctly 1`] = `
Object {
  "display": "none",
}
`;
